### PR TITLE
Incorporating SNP data

### PR DESCRIPTION
**Pull request by rnmitchell**
This PR will be for including SNP data into lusSTR for ForenSeq. There are several steps involved with this:
* Creating new commands specific to SNPs (something like `format_snps` and `annotate_snps`)
* Formatting SNP data from either UAS Sample Details Report and Phenotype report or STRait Razor data
* Identifying SNPs reported on the reverse strand in UAS
* Identifying coordinate of SNP in STRait Razor sequence string (individual SNPs are not reported, only sequence string)
* Including identity SNPs, ancestry SNPs and phenotype SNPS (have argument to specify which SNPs to include)

### CODE REVIEW

**Comment by rnmitchell** [comment]
I changed the original `format` code to do a quick check that that only STRait Razor .txt files are included in the compiled output file. I thought it might be advantageous to be able to have both the original STRait Razor output files and lusSTR output files all in the same directory (and can run lusSTR multiple times). It will exclude .txt files which throw a ValueError (caused by having final lusSTR output files within the provided directory).

**Comment by rnmitchell** [comment]
I noticed the UAS Sample Details Reports used in the UAS_bulk_upload tests contained sample/analysis/project IDs. I changed the files to positive control (2800M) UAS Sample Detail Reports (and gave generic IDs). Also fixed the `format.py` script to not duplicate the provided directory name in the code (I have no idea how this worked before??).

**Comment by standage** [comment]
> I have no idea how this worked before??

Ahh, I see you've reached the final stage of programmer's grief.

1. That can't happen.
2. That doesn't happen on my machine.
3. Wait, that shouldn't happen!
4. Why does that happen?
5. Oh, I see...
6. How did that ever work?!?!

**Comment by rnmitchell** [comment]
@standage this PR is ready for review. Let me know if you think there are additional tests I should include.

I had mentioned in our meeting yesterday that we were getting additional data to test the code with, but I have no idea when that would be, so I think we can move ahead with this and make any changes based on new data later.

File: lusSTR/snps.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+    final table based on the specified SNP type from the CLI.
+    '''
+    if snpid == 'N29insA':
+        snpid = 'rs312262906_N29insA'
+    metadata = snp_marker_data[snpid]
+    snp_type = metadata['Type']
+    seq = infile.iloc[j, 2]
+    expected_alleles = metadata['Alleles']
+    snp_loc = metadata['Coord']
+    snp_call = seq[snp_loc]
```
Comment (standage) [inline lusSTR/snps.py:212]: So these two lines are the primary logic for determining the SNP allele, right? Everything else seems to be handling special cases or putative and unexpected indels.

**Review by standage — APPROVED** [review]
Looks good to me. I can't claim to have given every block of code a thorough examination. But the central logic seems pretty straightforward, the CLI is clear and well documented, and the tests pass for me, so thumbs up from me!

File: lusSTR/snps.py (hunk 1 above)
Comment (rnmitchell) [reply lusSTR/snps.py:212]: Yup!

**Comment by rnmitchell** [comment]
Can you approve the merge? It' still saying Review required for some reason.

**Comment by rnmitchell** [comment]
Oh... nevermind! I guess my page just had to refresh :)

**Pull request merged**